ext/dom/document.c

```diff
@@ -864,17 +869,11 @@ PHP_METHOD(DOMDocument, createElementNS)
 	}
 
 	if (errorcode != 0) {
-		if (nodep != NULL) {
-			xmlFreeNode(nodep);
-		}
+		xmlFreeNode(nodep);
 		php_dom_throw_error(errorcode, dom_get_strict_error(intern->document));
 		RETURN_FALSE;
```

Comment:
Probably makes sense to rewrite this code prior to the ``if(errorcode == 0)`` and use this as a guard as the only other case that seems to fail as far as I can see is on line 862.

Reply:
There's also `nsptr = dom_get_ns(nodep, uri, &errorcode, prefix);` that has an out-parameter that can set the `errorcode`. And on line 844 the error code can also be set initially. I think I'm going to keep it as-is because it has a single easy-to-spot error handling block for the `errorcode` this way.

Reply:
Right :+1: